src/Microsoft.ML.Api/DataViewConstructionUtils.cs

```diff
@@ -790,17 +791,17 @@ protected override bool MoveManyCore(long count)
             }
         }
 
-        internal static Schema.Column[] GetSchemaColumns(InternalSchemaDefinition schemaDefn)
+        internal static Schema.DetachedColumn[] GetSchemaColumns(InternalSchemaDefinition schemaDefn)
```

Comment:
>Schema.DetachedColumn[] GetSchemaColumns [](start = 24, length = 40)

This is one of examples why I don't like DetachedColumns as a name. 
We have Schema.Columns and we have Schema.DetachedColumns and based on this method I would expect it to return Schema.Columns.

Reply:
I think you are sort-of biased as an ML.NET dev. `Schema.Column` is truly user-facing concept: it's a column of the schema of the data.
`Schema.DetachedColumn` is a component-authoring concept: it's a 'pre-column' not yet associated with schema. It's like `Schema.Column`, but with some properties removed.
Specifically, methods that return `DetachedColumn` should not be called by pipeline author: we would even make them internal if we could.

---
In reply to: [235482239](https://github.com/dotnet/machinelearning/pull/1680#discussion_r235482239) [](ancestors = 235482239)